dot.cpp

```diff
@@ -107,6 +150,21 @@ void Dot::newFoodSource(pair<uint16_t, uint16_t> food_pos, size_t idx, uint16_t
   }
 }
 
+void Dot::newPrey(pair<uint16_t, uint16_t> prey_pos, size_t idx, uint16_t dist){
+  if(!prey_in_sight_ || dist < prey_in_sight_dist_){
+    prey_in_sight_ = true;
+    prey_in_sight_pos_ = prey_pos;
+    prey_in_sight_idx_ = idx;
+    prey_in_sight_dist_ = dist;
+  }
+}
+
+void Dot::newHazardSource(pair<uint16_t, uint16_t> hazard_pos, uint16_t dist){
+  hazard_in_sight_ = true;
+  hazard_in_sight_pos_ = hazard_pos;
+  hazard_in_sight_dist_ = dist;
+}
```

Comment:
If a Dots receives information on a new hazard source, it forgets its previously known hazard source. We should either add a check for distances: A Dot only stores the closest hazard, or work with a vector. The later could cause some difficulties when determining the direction in which the Dot should run.